lib/ansible/modules/notification/catapult.py

```diff
@@ -0,0 +1,165 @@
+    required: true
+    default: null
+  dest:
+    description:
+      - The phone number or numbers the message should be sent to (must be in E.164 format, like +19195551212).
+    required: true
+    default: null
+  msg:
+    description:
+      - The contents of the text message (must be 2048 characters or less).
+    required: true
+    default: null
+  media:
+    description:
+      - For MMS messages, a media url to the location of the media to be sent with the message.
+    required: false
+    default: null
+  user_id:
+    description:
+      - User Id from Api account page
+    required: true
+    default: null
+  api_token: 
+    description:
+      - Api Token from Api account page
+    required: true
+    default: null
+  api_secret: 
+    description:
+      - Api Secret from Api account page
+    required: true
+    default: null
+
+author: "Jonathan Mainguy (@Jmainguy)"
+notes:
+    - Will return changed even if the media url is wrong
+    - Will return changed if the destination number is invalid
+
+'''
+
+EXAMPLES = '''
+# Send a mms to multiple users
+- catapult: 
+  src: "+15035555555" 
+  dest: 
+    - "+12525089000"
+    - "+12018994225"
+  media: "https://pbs.twimg.com/profile_images/2789370216/8765b6ba61039a987bdc1b3bc922bdbf_400x400.png"
+  msg: "Task is complete"
+  user_id: "{{ user_id }}"
+  api_token: "{{ api_token }}"
+  api_secret: "{{ api_secret }}"
+
+# Send a sms to a single user
+- catapult: 
+  src: "+15035555555" 
+  dest: "+12018994225"
+  msg: "Consider yourself notified"
+  user_id: "{{ user_id }}"
+  api_token: "{{ api_token }}"
+  api_secret: "{{ api_secret }}"
+
+'''
+
+RETURN = '''
+changed:
+    description: Whether the api accepted the message.
+    returned: always
+    type: boolean
+    sample: True
+'''
+
+
+import json
+# Stole most of this code from the ansible twillio and nexmo modules
+
+def send(module, src, dest, msg, media, user_id, api_token, api_secret):
+    """
+    Send the message
+    """
+    AGENT = "Ansible"
+    URI = "https://api.catapult.inetwork.com/v1/users/%s/messages" % user_id
+    data = {'from':src, 'to':dest, 'text':msg}
+    if media:
+        data['media'] = media
+
+    headers = {'User-Agent': AGENT,
+            'Content-type': 'application/json',
+            }
+
+    # Hack module params to have the Basic auth params that fetch_url expects
+    module.params['url_username'] = api_token.replace('\n', '')
+    module.params['url_password'] = api_secret.replace('\n', '')
+
+    return fetch_url(module, URI, data=json.dumps(data), headers=headers, method="post")
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            src               = dict(required=True),
+            dest              = dict(required=True, type='list'),
+            msg               = dict(required=True),
+            user_id           = dict(required=True, no_log=True),
```

Comment:
Its part of the credentials to login, I have no problem removing the no_log=True from it though.